main.go

```diff
@@ -162,47 +133,105 @@ func main() {
 			return err
 		}
 		return nil
-	}); err != nil {
-		log.Errorf("Nuget restore failed, error: %s", err)
-		os.Exit(1)
+	})
+}
+
+// collectCaches collects the caches based on the allowed config.
+// For more information about caches please read: https://docs.microsoft.com/en-us/nuget/consume-packages/managing-the-global-packages-and-cache-folders
+func collectCaches(allowCaching string) error {
+	nuGetCache := cache.New()
+	switch allowCaching {
+	case cacheInputNone:
+		return nil
+	case cacheInputlocal:
+		collectLocalCaches(&nuGetCache)
+	case cacheInputGlobal:
+		collectGlobalCaches(&nuGetCache)
 	}
+	return nuGetCache.Commit()
+}
 
-	// Collecting caches
-	fmt.Println()
-	log.Infof("Collecting NuGet cache...")
+// collectHTTPCaches collects the HTTP cache.
+func collectHTTPCaches(nuGetCache *cache.Cache) {
+	httpCachePth := getHTTPCachePath()
+	if exists, err := pathutil.IsPathExists(httpCachePth); err != nil {
+		log.Warnf("Failed to determine if path (%s) exists, error: %s", httpCachePth, err)
+	} else if exists {
+		nuGetCache.IncludePath(httpCachePth)
+	}
+}
 
-	nugetCache := cache.New()
+// getHTTPCachePath gets the path for the HTTP cache.
+func getHTTPCachePath() string {
```

Comment:
do not use `get` prefix: https://golang.org/doc/effective_go.html#Getters

Reply:
Removed prefix.